src/lex.js

```diff
@@ -660,19 +704,21 @@ Lexer.prototype = {
     }.bind(this);
 
     function removeEscapeSequences(id) {
-      return id.replace(/\\u([0-9a-fA-F]{4})/g, function(m0, codepoint) {
-        return String.fromCharCode(parseInt(codepoint, 16));
+      return id.replace(/\\u(?:([0-9a-fA-F]{4})|\{([0-9a-fA-F]{1,6})\})/g, function(m0, cp1, cp2) {
```

Comment:
This pattern is now probably too complicated to simply declare in-line. Mind moving it to `src/reg.js`?